nan.h

```diff
@@ -644,73 +644,83 @@ return args.GetReturnValue().Set(Nan::imp::NanEnsureHandleOrPersistent(value))
       info_->persistent.SetWeak(info_, &_NanWeakCallbackDispatcher<T, P>);
   }
 
-template<typename T, typename P>
-NAN_INLINE _NanWeakCallbackInfo<T, P>* NanMakeWeakPersistent(
-    v8::Handle<T> handle
-  , P* parameter
-  , typename _NanWeakCallbackInfo<T, P>::Callback callback) {
-    _NanWeakCallbackInfo<T, P> *cbinfo =
-     new _NanWeakCallbackInfo<T, P>(handle, parameter, callback);
-    cbinfo->persistent.SetWeak(cbinfo, &_NanWeakCallbackDispatcher<T, P>);
-    return cbinfo;
-}
+  template<typename T, typename P>
+  NAN_INLINE _NanWeakCallbackInfo<T, P>* NanMakeWeakPersistent(
+      v8::Handle<T> handle
+    , P* parameter
+    , typename _NanWeakCallbackInfo<T, P>::Callback callback) {
+      _NanWeakCallbackInfo<T, P> *cbinfo =
+       new _NanWeakCallbackInfo<T, P>(handle, parameter, callback);
+      cbinfo->persistent.SetWeak(cbinfo, &_NanWeakCallbackDispatcher<T, P>);
+      return cbinfo;
+  }
 
 # define NAN_WEAK_CALLBACK(name)                                               \
     template<typename T, typename P>                                           \
     static void name(const _NanWeakCallbackData<T, P> &data)
 
-# define _NAN_ERROR(fun, errmsg) fun(NanNew<v8::String>(errmsg))
-
-# define _NAN_THROW_ERROR(fun, errmsg)                                         \
-    do {                                                                       \
+# define ERROR_TYPES                                                           \
```

Comment:
Suggestion: name this NAN_ERROR_TYPES just in case a type/value/macro ERROR_TYPES exists elsewhere.  Or scrap it and call X(Error), X(RangeError), etc. directly below.